### PR TITLE
Card Class Test Cases

### PR DESCRIPTION
**Pull request by iamalotaibi**
adding test cases for Card class that coverage 100%

### CODE REVIEW

**Comment by janzeng** [comment]
Would it work to set (cards_left == 4) inside inEndState() to get 100% line coverage for Game? If not everything else looks great!

File: src/test/java/models/testGame.java [hunk 1]
```diff
@@ -212,4 +212,9 @@ public void testWinLoose() {
         assertEquals(false, g2.isCardMovable(2));
         assertEquals(false, g2.isCardMovable(3));
     }
+    @Test
```
Comment (janzeng) [inline src/test/java/models/testGame.java]: This test is great! It covers everything needed to verify that the getScore method is working correctly.

Comment (iamalotaibi) [reply src/test/java/models/testGame.java]: I have made another pull request for game class, this line will be fixed if you accepted this first then the other pull request. Good catch!

File: src/test/java/models/testGame.java [hunk 2]
```diff
@@ -212,4 +212,9 @@ public void testWinLoose() {
...
+    public void testUpdateGameEndState() {
+        AcesUpGame g1 = new AcesUpGame();
+        assertEquals(0, g1.getScore());
```
Comment (AntonSynytsiaOSU) [inline src/test/java/models/testGame.java]: This covers the getScore function but an additional test that ensures the score isn't zero, for instance, after the first remove, would cover the functionality. This is how it could look:
```
// Deal out all the cards
for (int i = 0; i < 13; ++i) {
    g1.dealFour();
}
// Attempt removing a card and checking the next score is not zero
for (int i = 0; i < 4; ++i) {
    if (g1.isCardRemovable(i)) {
        g1.remove(i);
        assertNotEquals(0, g1.getScore());
        break;
    }
}
```

**Comment by janzeng** [comment]
Awesome! Looks great!
